base/repl/LineEdit.jl

```diff
@@ -970,15 +967,15 @@ function write_response_buffer(s::PromptState, data)
 end
 
 mutable struct SearchState <: ModeState
-    terminal
-    histprompt
+    terminal::AbstractTerminal
+    histprompt # ::HistoryPrompt
```

Comment:
Because that's the type in practice, but `HistoryPrompt` doesn't exist yet when this is defined. The reverse dependency is unfortunate and something that should be fixed, but that kind of refactoring is beyond the scope of this PR.